pmd-core/src/main/java/net/sourceforge/pmd/lang/rule/properties/AbstractMultiNumericProperty.java

```diff
@@ -4,48 +4,103 @@
+    implements NumericPropertyDescriptor<List<T>> {
+
+    private final T lowerLimit;
+    private final T upperLimit;
+
 
     /**
-     * Constructor for AbstractMultiNumericProperty.
+     * Constructor for a multi-valued numeric property using a list of defaults.
+     *
+     * @param theName        Name
+     * @param theDescription Description
+     * @param lower          Minimum value of the property
+     * @param upper          Maximum value of the property
+     * @param theDefault     List of defaults
+     * @param theUIOrder     UI order
      *
-     * @param theName
-     *            String
-     * @param theDescription
-     *            String
-     * @param lower
-     *            Number
-     * @param upper
-     *            Number
-     * @param theDefault
-     *            T
-     * @param theUIOrder
-     *            float
+     * @throws IllegalArgumentException if lower > upper, or one of them is null, or one of the defaults is not between
+     * the bounds
      */
-    protected AbstractMultiNumericProperty(String theName, String theDescription, Number lower, Number upper,
-            T theDefault, float theUIOrder) {
-        super(theName, theDescription, lower, upper, theDefault, theUIOrder);
+    AbstractMultiNumericProperty(String theName, String theDescription, T lower, T upper, List<T> theDefault,
+                                 float theUIOrder) {
+        super(theName, theDescription, theDefault, theUIOrder);
+
+        lowerLimit = lower;
+        upperLimit = upper;
+
+        checkNumber(lower);
+        checkNumber(upper);
+
+        if (lower.doubleValue() > upper.doubleValue()) {
+            throw new IllegalArgumentException("Lower limit cannot be greater than the upper limit");
+        }
+
+
     }
 
-    /**
-     * @return boolean
-     * @see net.sourceforge.pmd.PropertyDescriptor#isMultiValue()
-     */
+
+    private void checkNumber(T number) {
+        if (valueErrorFor(number) != null) {
+            throw new IllegalArgumentException(valueErrorFor(number));
+        }
+    }
+
+
     @Override
-    public boolean isMultiValue() {
-        return true;
+    protected String valueErrorFor(T value) {
+
+        double number = value.doubleValue();
+
+        if (number > upperLimit.doubleValue() || number < lowerLimit.doubleValue()) {
+            return value + " is out of range " + AbstractNumericProperty.rangeString(lowerLimit, upperLimit);
+        }
+
+        return null;
     }
 
-    /**
-     * @return String
-     */
+
+    @Override
+    public Number lowerLimit() {
+        return lowerLimit;
+    }
+
+
+    @Override
+    public Number upperLimit() {
+        return upperLimit;
+    }
+
+
+    @Override
+    protected void addAttributesTo(Map<PropertyDescriptorField, String> attributes) {
+        super.addAttributesTo(attributes);
+        attributes.put(MIN, lowerLimit.toString());
```

Comment:
I've no decision made yet, when we should upgrading to java8.
Looking at maven and gradle, as the two most common build tools, they both are still on java7 as a minimum requirement. Once they upgrade, it doesn't make sense for us, to stay behind. But until then: Upgrading PMD to java8 would force the end users to use java8, too (effectively changing the minimum required java version of maven/gradle to java8 then). No big deal personally, and thanks to the toolchains support in maven, it's easy to run maven with java8, but let the code be compiled with a real java7 compiler (no -target flags). Not sure, how easy such a mix is with gradle.
The users, who are still on 1.7 (or even 1.6), I assume, are bigger companies, that move a bit slower. I know, that [tiobe](https://github.com/tiobe/pmd) is maintaining PMD 5.3.x for customers, that are still on java 1.6...

Looking at the current maven plugins, no common/official plugin is yet moved to java8 (see the [profiles java-6 and java-7](https://github.com/apache/maven-plugins/blob/trunk/pom.xml#L111)).
Upgrading to java8 has at least the follow-up to upgrade maven-pmd-plugin to java8, too.

So, for now, I would say, we post-pone this decision. Implementing it would be anyway a completely independent task with no functional change.

For completeness, that's [the oracle support roadmap](http://www.oracle.com/technetwork/java/eol-135779.html):
* Java 7: Available since: July 2011, end of public updates: April 2015 (more than 2 years ago...)
* Java 8: Available since: March 2014, end of public updates: probably September 2018 (in about 1 year...)